nixos/modules/virtualisation/oci-image.nix

```diff
@@ -0,0 +1,48 @@
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  imports = [ ./oci-common.nix ];
+
+  config = {
+    system.build.OCIImage = import ../../lib/make-disk-image.nix {
+      inherit config lib pkgs;
+      name = "oci-image";
+      configFile = ./oci-config-user.nix;
+      format = "qcow2";
+      diskSize = 8192;
+    };
+
+    systemd.services.fetch-ssh-keys = {
+      description = "Fetch authorized_keys for root user";
+
+      wantedBy = [ "sshd.service" ];
+      before = [ "sshd.service" ];
+
+      after = [ "network-online.target" ];
+      wants = [ "network-online.target" ];
+
+      path  = [ pkgs.coreutils pkgs.curl ];
```

Comment:
Should this service be enabled only on the very first profile? I wonder if this should be part of the -user config.

Reply:
The idea here is that the service is executed at least once before the initial ssh connection. This service does indeed need to only run once, since the ssh keys can't be modified in OCI after creating the instance AFAIK.
It should indeed be enabled only for the first profile. The user is allowed to change the system configuration to not include the `fetch-ssh-keys` service, since all ssh keys should be present in `/root/.ssh/authorized_keys` by then, hence this service is not included in `oci-config-user.nix`.